{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Let's validate the model using DRUM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/bin/drum\", line 6, in <module>\n",
      "    main()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/main.py\", line 96, in main\n",
      "    CMRunner(runtime).run()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 235, in run\n",
      "    self._run_fit_and_predictions_pipelines_in_mlpiper()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 470, in _run_fit_and_predictions_pipelines_in_mlpiper\n",
      "    _pipeline_executor.run_pipeline(cleanup=False)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 261, in run_pipeline\n",
      "    self._run_pipeline()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 251, in _run_pipeline\n",
      "    self._dag.run_connected_pipeline(self._ml_engine)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/dag.py\", line 138, in run_connected_pipeline\n",
      "    data_objs = dag_node.component_runner.run(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py\", line 14, in run\n",
      "    data_objs = self._dag_node.main_cls().materialize(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/components/connectable_component.py\", line 11, in materialize\n",
      "    return self._materialize(parent_data_objs, self._ml_engine.user_data)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py\", line 51, in _materialize\n",
      "    predictions = self._predictor.predict(df)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py\", line 37, in predict\n",
      "    predictions = self._model_adapter.predict(data=df, model=self._model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 299, in predict\n",
      "    ) from None\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 295, in predict\n",
      "    predictions = self._predictor_to_use.predict(data, model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py\", line 92, in predict\n",
      "    [float(prediction) for prediction in model.predict(data)],\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 783, in predict\n",
      "    X = self._validate_X_predict(X)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 421, in _validate_X_predict\n",
      "    return self.estimators_[0]._validate_X_predict(X, check_input=True)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/tree/_classes.py\", line 388, in _validate_X_predict\n",
      "    X = check_array(X, dtype=DTYPE, accept_sparse=\"csr\")\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 73, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 646, in check_array\n",
      "    allow_nan=force_all_finite == 'allow-nan')\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 100, in _assert_all_finite\n",
      "    msg_dtype if msg_dtype is not None else X.dtype)\n",
      "ValueError: Failure when making predictions: Input contains NaN, infinity or a value too large for dtype('float32').\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/bin/drum\", line 6, in <module>\n",
      "    main()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/main.py\", line 96, in main\n",
      "    CMRunner(runtime).run()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 235, in run\n",
      "    self._run_fit_and_predictions_pipelines_in_mlpiper()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 470, in _run_fit_and_predictions_pipelines_in_mlpiper\n",
      "    _pipeline_executor.run_pipeline(cleanup=False)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 261, in run_pipeline\n",
      "    self._run_pipeline()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 251, in _run_pipeline\n",
      "    self._dag.run_connected_pipeline(self._ml_engine)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/dag.py\", line 138, in run_connected_pipeline\n",
      "    data_objs = dag_node.component_runner.run(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py\", line 14, in run\n",
      "    data_objs = self._dag_node.main_cls().materialize(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/components/connectable_component.py\", line 11, in materialize\n",
      "    return self._materialize(parent_data_objs, self._ml_engine.user_data)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py\", line 51, in _materialize\n",
      "    predictions = self._predictor.predict(df)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py\", line 37, in predict\n",
      "    predictions = self._model_adapter.predict(data=df, model=self._model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 299, in predict\n",
      "    ) from None\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 295, in predict\n",
      "    predictions = self._predictor_to_use.predict(data, model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py\", line 92, in predict\n",
      "    [float(prediction) for prediction in model.predict(data)],\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 783, in predict\n",
      "    X = self._validate_X_predict(X)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 421, in _validate_X_predict\n",
      "    return self.estimators_[0]._validate_X_predict(X, check_input=True)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/tree/_classes.py\", line 388, in _validate_X_predict\n",
      "    X = check_array(X, dtype=DTYPE, accept_sparse=\"csr\")\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 73, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 646, in check_array\n",
      "    allow_nan=force_all_finite == 'allow-nan')\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 100, in _assert_all_finite\n",
      "    msg_dtype if msg_dtype is not None else X.dtype)\n",
      "ValueError: Failure when making predictions: Input contains NaN, infinity or a value too large for dtype('float32').\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/bin/drum\", line 6, in <module>\n",
      "    main()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/main.py\", line 96, in main\n",
      "    CMRunner(runtime).run()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 235, in run\n",
      "    self._run_fit_and_predictions_pipelines_in_mlpiper()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 470, in _run_fit_and_predictions_pipelines_in_mlpiper\n",
      "    _pipeline_executor.run_pipeline(cleanup=False)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 261, in run_pipeline\n",
      "    self._run_pipeline()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 251, in _run_pipeline\n",
      "    self._dag.run_connected_pipeline(self._ml_engine)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/dag.py\", line 138, in run_connected_pipeline\n",
      "    data_objs = dag_node.component_runner.run(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py\", line 14, in run\n",
      "    data_objs = self._dag_node.main_cls().materialize(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/components/connectable_component.py\", line 11, in materialize\n",
      "    return self._materialize(parent_data_objs, self._ml_engine.user_data)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py\", line 51, in _materialize\n",
      "    predictions = self._predictor.predict(df)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py\", line 37, in predict\n",
      "    predictions = self._model_adapter.predict(data=df, model=self._model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 299, in predict\n",
      "    ) from None\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 295, in predict\n",
      "    predictions = self._predictor_to_use.predict(data, model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py\", line 92, in predict\n",
      "    [float(prediction) for prediction in model.predict(data)],\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 783, in predict\n",
      "    X = self._validate_X_predict(X)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 421, in _validate_X_predict\n",
      "    return self.estimators_[0]._validate_X_predict(X, check_input=True)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/tree/_classes.py\", line 388, in _validate_X_predict\n",
      "    X = check_array(X, dtype=DTYPE, accept_sparse=\"csr\")\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 73, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 646, in check_array\n",
      "    allow_nan=force_all_finite == 'allow-nan')\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 100, in _assert_all_finite\n",
      "    msg_dtype if msg_dtype is not None else X.dtype)\n",
      "ValueError: Failure when making predictions: Input contains NaN, infinity or a value too large for dtype('float32').\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/bin/drum\", line 6, in <module>\n",
      "    main()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/main.py\", line 96, in main\n",
      "    CMRunner(runtime).run()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 235, in run\n",
      "    self._run_fit_and_predictions_pipelines_in_mlpiper()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 470, in _run_fit_and_predictions_pipelines_in_mlpiper\n",
      "    _pipeline_executor.run_pipeline(cleanup=False)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 261, in run_pipeline\n",
      "    self._run_pipeline()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 251, in _run_pipeline\n",
      "    self._dag.run_connected_pipeline(self._ml_engine)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/dag.py\", line 138, in run_connected_pipeline\n",
      "    data_objs = dag_node.component_runner.run(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py\", line 14, in run\n",
      "    data_objs = self._dag_node.main_cls().materialize(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/components/connectable_component.py\", line 11, in materialize\n",
      "    return self._materialize(parent_data_objs, self._ml_engine.user_data)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py\", line 51, in _materialize\n",
      "    predictions = self._predictor.predict(df)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py\", line 37, in predict\n",
      "    predictions = self._model_adapter.predict(data=df, model=self._model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 299, in predict\n",
      "    ) from None\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 295, in predict\n",
      "    predictions = self._predictor_to_use.predict(data, model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py\", line 92, in predict\n",
      "    [float(prediction) for prediction in model.predict(data)],\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 783, in predict\n",
      "    X = self._validate_X_predict(X)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 421, in _validate_X_predict\n",
      "    return self.estimators_[0]._validate_X_predict(X, check_input=True)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/tree/_classes.py\", line 388, in _validate_X_predict\n",
      "    X = check_array(X, dtype=DTYPE, accept_sparse=\"csr\")\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 73, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 646, in check_array\n",
      "    allow_nan=force_all_finite == 'allow-nan')\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 100, in _assert_all_finite\n",
      "    msg_dtype if msg_dtype is not None else X.dtype)\n",
      "ValueError: Failure when making predictions: Input contains NaN, infinity or a value too large for dtype('float32').\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/bin/drum\", line 6, in <module>\n",
      "    main()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/main.py\", line 96, in main\n",
      "    CMRunner(runtime).run()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 235, in run\n",
      "    self._run_fit_and_predictions_pipelines_in_mlpiper()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 470, in _run_fit_and_predictions_pipelines_in_mlpiper\n",
      "    _pipeline_executor.run_pipeline(cleanup=False)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 261, in run_pipeline\n",
      "    self._run_pipeline()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 251, in _run_pipeline\n",
      "    self._dag.run_connected_pipeline(self._ml_engine)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/dag.py\", line 138, in run_connected_pipeline\n",
      "    data_objs = dag_node.component_runner.run(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py\", line 14, in run\n",
      "    data_objs = self._dag_node.main_cls().materialize(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/components/connectable_component.py\", line 11, in materialize\n",
      "    return self._materialize(parent_data_objs, self._ml_engine.user_data)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py\", line 51, in _materialize\n",
      "    predictions = self._predictor.predict(df)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py\", line 37, in predict\n",
      "    predictions = self._model_adapter.predict(data=df, model=self._model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 299, in predict\n",
      "    ) from None\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 295, in predict\n",
      "    predictions = self._predictor_to_use.predict(data, model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py\", line 92, in predict\n",
      "    [float(prediction) for prediction in model.predict(data)],\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 783, in predict\n",
      "    X = self._validate_X_predict(X)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 421, in _validate_X_predict\n",
      "    return self.estimators_[0]._validate_X_predict(X, check_input=True)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/tree/_classes.py\", line 388, in _validate_X_predict\n",
      "    X = check_array(X, dtype=DTYPE, accept_sparse=\"csr\")\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 73, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 646, in check_array\n",
      "    allow_nan=force_all_finite == 'allow-nan')\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 100, in _assert_all_finite\n",
      "    msg_dtype if msg_dtype is not None else X.dtype)\n",
      "ValueError: Failure when making predictions: Input contains NaN, infinity or a value too large for dtype('float32').\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/bin/drum\", line 6, in <module>\n",
      "    main()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/main.py\", line 96, in main\n",
      "    CMRunner(runtime).run()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 235, in run\n",
      "    self._run_fit_and_predictions_pipelines_in_mlpiper()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 470, in _run_fit_and_predictions_pipelines_in_mlpiper\n",
      "    _pipeline_executor.run_pipeline(cleanup=False)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 261, in run_pipeline\n",
      "    self._run_pipeline()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 251, in _run_pipeline\n",
      "    self._dag.run_connected_pipeline(self._ml_engine)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/dag.py\", line 138, in run_connected_pipeline\n",
      "    data_objs = dag_node.component_runner.run(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py\", line 14, in run\n",
      "    data_objs = self._dag_node.main_cls().materialize(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/components/connectable_component.py\", line 11, in materialize\n",
      "    return self._materialize(parent_data_objs, self._ml_engine.user_data)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py\", line 51, in _materialize\n",
      "    predictions = self._predictor.predict(df)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py\", line 37, in predict\n",
      "    predictions = self._model_adapter.predict(data=df, model=self._model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 299, in predict\n",
      "    ) from None\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 295, in predict\n",
      "    predictions = self._predictor_to_use.predict(data, model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py\", line 92, in predict\n",
      "    [float(prediction) for prediction in model.predict(data)],\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 783, in predict\n",
      "    X = self._validate_X_predict(X)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 421, in _validate_X_predict\n",
      "    return self.estimators_[0]._validate_X_predict(X, check_input=True)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/tree/_classes.py\", line 388, in _validate_X_predict\n",
      "    X = check_array(X, dtype=DTYPE, accept_sparse=\"csr\")\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 73, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 646, in check_array\n",
      "    allow_nan=force_all_finite == 'allow-nan')\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 100, in _assert_all_finite\n",
      "    msg_dtype if msg_dtype is not None else X.dtype)\n",
      "ValueError: Failure when making predictions: Input contains NaN, infinity or a value too large for dtype('float32').\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/bin/drum\", line 6, in <module>\n",
      "    main()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/main.py\", line 96, in main\n",
      "    CMRunner(runtime).run()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 235, in run\n",
      "    self._run_fit_and_predictions_pipelines_in_mlpiper()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 470, in _run_fit_and_predictions_pipelines_in_mlpiper\n",
      "    _pipeline_executor.run_pipeline(cleanup=False)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 261, in run_pipeline\n",
      "    self._run_pipeline()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 251, in _run_pipeline\n",
      "    self._dag.run_connected_pipeline(self._ml_engine)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/dag.py\", line 138, in run_connected_pipeline\n",
      "    data_objs = dag_node.component_runner.run(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py\", line 14, in run\n",
      "    data_objs = self._dag_node.main_cls().materialize(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/components/connectable_component.py\", line 11, in materialize\n",
      "    return self._materialize(parent_data_objs, self._ml_engine.user_data)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py\", line 51, in _materialize\n",
      "    predictions = self._predictor.predict(df)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py\", line 37, in predict\n",
      "    predictions = self._model_adapter.predict(data=df, model=self._model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 299, in predict\n",
      "    ) from None\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 295, in predict\n",
      "    predictions = self._predictor_to_use.predict(data, model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py\", line 92, in predict\n",
      "    [float(prediction) for prediction in model.predict(data)],\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 783, in predict\n",
      "    X = self._validate_X_predict(X)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 421, in _validate_X_predict\n",
      "    return self.estimators_[0]._validate_X_predict(X, check_input=True)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/tree/_classes.py\", line 388, in _validate_X_predict\n",
      "    X = check_array(X, dtype=DTYPE, accept_sparse=\"csr\")\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 73, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 646, in check_array\n",
      "    allow_nan=force_all_finite == 'allow-nan')\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 100, in _assert_all_finite\n",
      "    msg_dtype if msg_dtype is not None else X.dtype)\n",
      "ValueError: Failure when making predictions: Input contains NaN, infinity or a value too large for dtype('float32').\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/bin/drum\", line 6, in <module>\n",
      "    main()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/main.py\", line 96, in main\n",
      "    CMRunner(runtime).run()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 235, in run\n",
      "    self._run_fit_and_predictions_pipelines_in_mlpiper()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 470, in _run_fit_and_predictions_pipelines_in_mlpiper\n",
      "    _pipeline_executor.run_pipeline(cleanup=False)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 261, in run_pipeline\n",
      "    self._run_pipeline()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 251, in _run_pipeline\n",
      "    self._dag.run_connected_pipeline(self._ml_engine)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/dag.py\", line 138, in run_connected_pipeline\n",
      "    data_objs = dag_node.component_runner.run(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py\", line 14, in run\n",
      "    data_objs = self._dag_node.main_cls().materialize(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/components/connectable_component.py\", line 11, in materialize\n",
      "    return self._materialize(parent_data_objs, self._ml_engine.user_data)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py\", line 51, in _materialize\n",
      "    predictions = self._predictor.predict(df)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py\", line 37, in predict\n",
      "    predictions = self._model_adapter.predict(data=df, model=self._model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 299, in predict\n",
      "    ) from None\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 295, in predict\n",
      "    predictions = self._predictor_to_use.predict(data, model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py\", line 92, in predict\n",
      "    [float(prediction) for prediction in model.predict(data)],\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 783, in predict\n",
      "    X = self._validate_X_predict(X)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 421, in _validate_X_predict\n",
      "    return self.estimators_[0]._validate_X_predict(X, check_input=True)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/tree/_classes.py\", line 388, in _validate_X_predict\n",
      "    X = check_array(X, dtype=DTYPE, accept_sparse=\"csr\")\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 73, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 646, in check_array\n",
      "    allow_nan=force_all_finite == 'allow-nan')\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 100, in _assert_all_finite\n",
      "    msg_dtype if msg_dtype is not None else X.dtype)\n",
      "ValueError: Failure when making predictions: Input contains NaN, infinity or a value too large for dtype('float32').\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/bin/drum\", line 6, in <module>\n",
      "    main()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/main.py\", line 96, in main\n",
      "    CMRunner(runtime).run()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 235, in run\n",
      "    self._run_fit_and_predictions_pipelines_in_mlpiper()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 470, in _run_fit_and_predictions_pipelines_in_mlpiper\n",
      "    _pipeline_executor.run_pipeline(cleanup=False)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 261, in run_pipeline\n",
      "    self._run_pipeline()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 251, in _run_pipeline\n",
      "    self._dag.run_connected_pipeline(self._ml_engine)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/dag.py\", line 138, in run_connected_pipeline\n",
      "    data_objs = dag_node.component_runner.run(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py\", line 14, in run\n",
      "    data_objs = self._dag_node.main_cls().materialize(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/components/connectable_component.py\", line 11, in materialize\n",
      "    return self._materialize(parent_data_objs, self._ml_engine.user_data)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py\", line 51, in _materialize\n",
      "    predictions = self._predictor.predict(df)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py\", line 37, in predict\n",
      "    predictions = self._model_adapter.predict(data=df, model=self._model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 299, in predict\n",
      "    ) from None\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 295, in predict\n",
      "    predictions = self._predictor_to_use.predict(data, model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py\", line 92, in predict\n",
      "    [float(prediction) for prediction in model.predict(data)],\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 783, in predict\n",
      "    X = self._validate_X_predict(X)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 421, in _validate_X_predict\n",
      "    return self.estimators_[0]._validate_X_predict(X, check_input=True)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/tree/_classes.py\", line 388, in _validate_X_predict\n",
      "    X = check_array(X, dtype=DTYPE, accept_sparse=\"csr\")\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 73, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 646, in check_array\n",
      "    allow_nan=force_all_finite == 'allow-nan')\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 100, in _assert_all_finite\n",
      "    msg_dtype if msg_dtype is not None else X.dtype)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ValueError: Failure when making predictions: Input contains NaN, infinity or a value too large for dtype('float32').\n",
      "    Predictions\n",
      "0        27.560\n",
      "1        30.030\n",
      "2        34.565\n",
      "3        34.120\n",
      "4        35.900\n",
      "5        29.920\n",
      "6        25.895\n",
      "7        22.710\n",
      "8        16.510\n",
      "9        20.135\n",
      "10       17.065\n",
      "11       23.370\n",
      "12       21.775\n",
      "13       26.775\n",
      "14       23.330\n",
      "15       26.755\n",
      "16       25.810\n",
      "17       19.915\n",
      "18       23.070\n",
      "19       21.980\n",
      "^C\n",
      "\n",
      "Ctrl+C pressed, aborting drum\n"
     ]
    }
   ],
   "source": [
    "!drum validation --code-dir ./custom_model --input data/boston_housing_test.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As you can see, we are getting an error that `ValueError: Failure when making predictions: Input contains NaN, infinity or a value too large for dtype('float32').`\n",
    "\n",
    "Basically, the model is not robust enough to handle null values and while our data does not include any null values, DRUM is checking automatically. To solve this, we are going to add a `custom.py` script within the `custom_model` folder which will fill null values if they are encountered."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    Predictions\n",
      "0        25.690\n",
      "1        21.850\n",
      "2        34.410\n",
      "3        33.730\n",
      "4        35.065\n",
      "5        25.735\n",
      "6        21.080\n",
      "7        22.570\n",
      "8        17.975\n",
      "9        20.095\n",
      "10       19.670\n",
      "11       19.300\n",
      "12       21.245\n",
      "13       19.790\n",
      "14       19.005\n",
      "15       19.710\n",
      "16       21.805\n",
      "17       18.965\n",
      "18       19.380\n",
      "19       18.595\n",
      "    Predictions\n",
      "0        25.630\n",
      "1        22.105\n",
      "2        34.410\n",
      "3        33.740\n",
      "4        35.625\n",
      "5        25.735\n",
      "6        21.830\n",
      "7        22.905\n",
      "8        16.465\n",
      "9        19.520\n",
      "10       17.250\n",
      "11       19.340\n",
      "12       21.180\n",
      "13       20.385\n",
      "14       18.335\n",
      "15       20.150\n",
      "16       22.530\n",
      "17       17.840\n",
      "18       19.575\n",
      "19       18.655\n",
      "    Predictions\n",
      "0        25.690\n",
      "1        22.025\n",
      "2        34.505\n",
      "3        33.720\n",
      "4        35.625\n",
      "5        25.735\n",
      "6        22.310\n",
      "7        22.570\n",
      "8        16.250\n",
      "9        19.715\n",
      "10       16.915\n",
      "11       20.360\n",
      "12       21.480\n",
      "13       20.740\n",
      "14       19.485\n",
      "15       20.640\n",
      "16       22.580\n",
      "17       18.240\n",
      "18       20.450\n",
      "19       20.295\n",
      "    Predictions\n",
      "0        25.690\n",
      "1        22.105\n",
      "2        34.410\n",
      "3        33.740\n",
      "4        35.625\n",
      "5        25.735\n",
      "6        21.830\n",
      "7        22.570\n",
      "8        16.250\n",
      "9        19.500\n",
      "10       16.915\n",
      "11       19.340\n",
      "12       21.245\n",
      "13       20.385\n",
      "14       18.335\n",
      "15       20.150\n",
      "16       22.530\n",
      "17       17.840\n",
      "18       19.575\n",
      "19       18.655\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/bin/drum\", line 6, in <module>\n",
      "    main()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/main.py\", line 96, in main\n",
      "    CMRunner(runtime).run()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 235, in run\n",
      "    self._run_fit_and_predictions_pipelines_in_mlpiper()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 470, in _run_fit_and_predictions_pipelines_in_mlpiper\n",
      "    _pipeline_executor.run_pipeline(cleanup=False)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 261, in run_pipeline\n",
      "    self._run_pipeline()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 251, in _run_pipeline\n",
      "    self._dag.run_connected_pipeline(self._ml_engine)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/dag.py\", line 138, in run_connected_pipeline\n",
      "    data_objs = dag_node.component_runner.run(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py\", line 14, in run\n",
      "    data_objs = self._dag_node.main_cls().materialize(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/components/connectable_component.py\", line 11, in materialize\n",
      "    return self._materialize(parent_data_objs, self._ml_engine.user_data)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py\", line 51, in _materialize\n",
      "    predictions = self._predictor.predict(df)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py\", line 37, in predict\n",
      "    predictions = self._model_adapter.predict(data=df, model=self._model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 299, in predict\n",
      "    ) from None\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 295, in predict\n",
      "    predictions = self._predictor_to_use.predict(data, model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py\", line 92, in predict\n",
      "    [float(prediction) for prediction in model.predict(data)],\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 783, in predict\n",
      "    X = self._validate_X_predict(X)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 421, in _validate_X_predict\n",
      "    return self.estimators_[0]._validate_X_predict(X, check_input=True)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/tree/_classes.py\", line 388, in _validate_X_predict\n",
      "    X = check_array(X, dtype=DTYPE, accept_sparse=\"csr\")\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 73, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 646, in check_array\n",
      "    allow_nan=force_all_finite == 'allow-nan')\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 100, in _assert_all_finite\n",
      "    msg_dtype if msg_dtype is not None else X.dtype)\n",
      "ValueError: Failure when making predictions: Input contains NaN, infinity or a value too large for dtype('float32').\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/bin/drum\", line 6, in <module>\n",
      "    main()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/main.py\", line 96, in main\n",
      "    CMRunner(runtime).run()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 235, in run\n",
      "    self._run_fit_and_predictions_pipelines_in_mlpiper()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 470, in _run_fit_and_predictions_pipelines_in_mlpiper\n",
      "    _pipeline_executor.run_pipeline(cleanup=False)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 261, in run_pipeline\n",
      "    self._run_pipeline()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 251, in _run_pipeline\n",
      "    self._dag.run_connected_pipeline(self._ml_engine)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/dag.py\", line 138, in run_connected_pipeline\n",
      "    data_objs = dag_node.component_runner.run(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py\", line 14, in run\n",
      "    data_objs = self._dag_node.main_cls().materialize(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/components/connectable_component.py\", line 11, in materialize\n",
      "    return self._materialize(parent_data_objs, self._ml_engine.user_data)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py\", line 51, in _materialize\n",
      "    predictions = self._predictor.predict(df)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py\", line 37, in predict\n",
      "    predictions = self._model_adapter.predict(data=df, model=self._model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 299, in predict\n",
      "    ) from None\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 295, in predict\n",
      "    predictions = self._predictor_to_use.predict(data, model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py\", line 92, in predict\n",
      "    [float(prediction) for prediction in model.predict(data)],\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 783, in predict\n",
      "    X = self._validate_X_predict(X)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 421, in _validate_X_predict\n",
      "    return self.estimators_[0]._validate_X_predict(X, check_input=True)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/tree/_classes.py\", line 388, in _validate_X_predict\n",
      "    X = check_array(X, dtype=DTYPE, accept_sparse=\"csr\")\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 73, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 646, in check_array\n",
      "    allow_nan=force_all_finite == 'allow-nan')\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 100, in _assert_all_finite\n",
      "    msg_dtype if msg_dtype is not None else X.dtype)\n",
      "ValueError: Failure when making predictions: Input contains NaN, infinity or a value too large for dtype('float32').\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/bin/drum\", line 6, in <module>\n",
      "    main()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/main.py\", line 96, in main\n",
      "    CMRunner(runtime).run()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 235, in run\n",
      "    self._run_fit_and_predictions_pipelines_in_mlpiper()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 470, in _run_fit_and_predictions_pipelines_in_mlpiper\n",
      "    _pipeline_executor.run_pipeline(cleanup=False)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 261, in run_pipeline\n",
      "    self._run_pipeline()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 251, in _run_pipeline\n",
      "    self._dag.run_connected_pipeline(self._ml_engine)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/dag.py\", line 138, in run_connected_pipeline\n",
      "    data_objs = dag_node.component_runner.run(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py\", line 14, in run\n",
      "    data_objs = self._dag_node.main_cls().materialize(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/components/connectable_component.py\", line 11, in materialize\n",
      "    return self._materialize(parent_data_objs, self._ml_engine.user_data)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py\", line 51, in _materialize\n",
      "    predictions = self._predictor.predict(df)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py\", line 37, in predict\n",
      "    predictions = self._model_adapter.predict(data=df, model=self._model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 299, in predict\n",
      "    ) from None\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 295, in predict\n",
      "    predictions = self._predictor_to_use.predict(data, model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py\", line 92, in predict\n",
      "    [float(prediction) for prediction in model.predict(data)],\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 783, in predict\n",
      "    X = self._validate_X_predict(X)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 421, in _validate_X_predict\n",
      "    return self.estimators_[0]._validate_X_predict(X, check_input=True)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/tree/_classes.py\", line 388, in _validate_X_predict\n",
      "    X = check_array(X, dtype=DTYPE, accept_sparse=\"csr\")\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 73, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 646, in check_array\n",
      "    allow_nan=force_all_finite == 'allow-nan')\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 100, in _assert_all_finite\n",
      "    msg_dtype if msg_dtype is not None else X.dtype)\n",
      "ValueError: Failure when making predictions: Input contains NaN, infinity or a value too large for dtype('float32').\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/bin/drum\", line 6, in <module>\n",
      "    main()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/main.py\", line 96, in main\n",
      "    CMRunner(runtime).run()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 235, in run\n",
      "    self._run_fit_and_predictions_pipelines_in_mlpiper()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 470, in _run_fit_and_predictions_pipelines_in_mlpiper\n",
      "    _pipeline_executor.run_pipeline(cleanup=False)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 261, in run_pipeline\n",
      "    self._run_pipeline()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 251, in _run_pipeline\n",
      "    self._dag.run_connected_pipeline(self._ml_engine)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/dag.py\", line 138, in run_connected_pipeline\n",
      "    data_objs = dag_node.component_runner.run(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py\", line 14, in run\n",
      "    data_objs = self._dag_node.main_cls().materialize(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/components/connectable_component.py\", line 11, in materialize\n",
      "    return self._materialize(parent_data_objs, self._ml_engine.user_data)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py\", line 51, in _materialize\n",
      "    predictions = self._predictor.predict(df)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py\", line 37, in predict\n",
      "    predictions = self._model_adapter.predict(data=df, model=self._model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 299, in predict\n",
      "    ) from None\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 295, in predict\n",
      "    predictions = self._predictor_to_use.predict(data, model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py\", line 92, in predict\n",
      "    [float(prediction) for prediction in model.predict(data)],\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 783, in predict\n",
      "    X = self._validate_X_predict(X)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 421, in _validate_X_predict\n",
      "    return self.estimators_[0]._validate_X_predict(X, check_input=True)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/tree/_classes.py\", line 388, in _validate_X_predict\n",
      "    X = check_array(X, dtype=DTYPE, accept_sparse=\"csr\")\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 73, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 646, in check_array\n",
      "    allow_nan=force_all_finite == 'allow-nan')\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 100, in _assert_all_finite\n",
      "    msg_dtype if msg_dtype is not None else X.dtype)\n",
      "ValueError: Failure when making predictions: Input contains NaN, infinity or a value too large for dtype('float32').\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/bin/drum\", line 6, in <module>\n",
      "    main()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/main.py\", line 96, in main\n",
      "    CMRunner(runtime).run()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 235, in run\n",
      "    self._run_fit_and_predictions_pipelines_in_mlpiper()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 470, in _run_fit_and_predictions_pipelines_in_mlpiper\n",
      "    _pipeline_executor.run_pipeline(cleanup=False)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 261, in run_pipeline\n",
      "    self._run_pipeline()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 251, in _run_pipeline\n",
      "    self._dag.run_connected_pipeline(self._ml_engine)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/dag.py\", line 138, in run_connected_pipeline\n",
      "    data_objs = dag_node.component_runner.run(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py\", line 14, in run\n",
      "    data_objs = self._dag_node.main_cls().materialize(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/components/connectable_component.py\", line 11, in materialize\n",
      "    return self._materialize(parent_data_objs, self._ml_engine.user_data)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py\", line 51, in _materialize\n",
      "    predictions = self._predictor.predict(df)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py\", line 37, in predict\n",
      "    predictions = self._model_adapter.predict(data=df, model=self._model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 299, in predict\n",
      "    ) from None\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 295, in predict\n",
      "    predictions = self._predictor_to_use.predict(data, model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py\", line 92, in predict\n",
      "    [float(prediction) for prediction in model.predict(data)],\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 783, in predict\n",
      "    X = self._validate_X_predict(X)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 421, in _validate_X_predict\n",
      "    return self.estimators_[0]._validate_X_predict(X, check_input=True)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/tree/_classes.py\", line 388, in _validate_X_predict\n",
      "    X = check_array(X, dtype=DTYPE, accept_sparse=\"csr\")\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 73, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 646, in check_array\n",
      "    allow_nan=force_all_finite == 'allow-nan')\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 100, in _assert_all_finite\n",
      "    msg_dtype if msg_dtype is not None else X.dtype)\n",
      "ValueError: Failure when making predictions: Input contains NaN, infinity or a value too large for dtype('float32').\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/bin/drum\", line 6, in <module>\n",
      "    main()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/main.py\", line 96, in main\n",
      "    CMRunner(runtime).run()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 235, in run\n",
      "    self._run_fit_and_predictions_pipelines_in_mlpiper()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/drum.py\", line 470, in _run_fit_and_predictions_pipelines_in_mlpiper\n",
      "    _pipeline_executor.run_pipeline(cleanup=False)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 261, in run_pipeline\n",
      "    self._run_pipeline()\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/executor.py\", line 251, in _run_pipeline\n",
      "    self._dag.run_connected_pipeline(self._ml_engine)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/dag.py\", line 138, in run_connected_pipeline\n",
      "    data_objs = dag_node.component_runner.run(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py\", line 14, in run\n",
      "    data_objs = self._dag_node.main_cls().materialize(parent_data_objs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/mlpiper/components/connectable_component.py\", line 11, in materialize\n",
      "    return self._materialize(parent_data_objs, self._ml_engine.user_data)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py\", line 51, in _materialize\n",
      "    predictions = self._predictor.predict(df)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py\", line 37, in predict\n",
      "    predictions = self._model_adapter.predict(data=df, model=self._model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 299, in predict\n",
      "    ) from None\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/model_adapter.py\", line 295, in predict\n",
      "    predictions = self._predictor_to_use.predict(data, model, **kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py\", line 92, in predict\n",
      "    [float(prediction) for prediction in model.predict(data)],\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 783, in predict\n",
      "    X = self._validate_X_predict(X)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/ensemble/_forest.py\", line 421, in _validate_X_predict\n",
      "    return self.estimators_[0]._validate_X_predict(X, check_input=True)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/tree/_classes.py\", line 388, in _validate_X_predict\n",
      "    X = check_array(X, dtype=DTYPE, accept_sparse=\"csr\")\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 73, in inner_f\n",
      "    return f(**kwargs)\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 646, in check_array\n",
      "    allow_nan=force_all_finite == 'allow-nan')\n",
      "  File \"/Users/thodoris.petropoulos/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py\", line 100, in _assert_all_finite\n",
      "    msg_dtype if msg_dtype is not None else X.dtype)\n",
      "ValueError: Failure when making predictions: Input contains NaN, infinity or a value too large for dtype('float32').\n",
      "^C\n",
      "\n",
      "Ctrl+C pressed, aborting drum\n"
     ]
    }
   ],
   "source": [
    "!drum validation --code-dir ./custom_model --input data/boston_housing_test.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "After moving the `custom.py` file inside the `custom_model` folder, we see that the results come back normally."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
